src/core/smp.cc

```diff
@@ -89,9 +91,18 @@ future<smp_service_group> create_smp_service_group(smp_service_group_config ssgc
 }
 
 future<> destroy_smp_service_group(smp_service_group ssg) noexcept {
-    return smp::submit_to(0, [ssg] {
-        return with_semaphore(smp_service_group_management_sem, 1, [ssg] {
-            auto id = internal::smp_service_group_id(ssg);
+    auto id = internal::smp_service_group_id(ssg);
+    return smp::submit_to(0, [id] {
+        return with_semaphore(smp_service_group_management_sem, 1, [id] {
+            if (id >= smp_service_groups.size()) {
+                auto msg = format("Invalid smp_service_group {}", id);
+                on_internal_error_noexcept(seastar_logger, msg);
+#ifndef SEASTAR_DEBUG
+                abort();
+#else
+                throw std::invalid_argument(msg);
+#endif
```

Comment:
Based on your reply on email to a similar patch for the smp service group:
```
On Sun, 2021-11-28 at 11:48 +0200, Avi Kivity wrote:

> This is similar to EBADF. If you can attempt to destroy an out-of-range 
> smp service group, you can attempt to destroy an in-use smp service 
> group that's unrelated to you. 
> Returning an exception is hiding a 
> critical error.
> 
> 
> Unrelated, destroy_smp_service_group is a close()-like function that is 
> used on shutdown paths. It's often not possible to recover from an error 
> there.
> 
> 
> I think on_internal_error_noexcept() is a better choice here.
```

However, this can't be tested if we abort uncoditionally, hence the different behavior in debug mode, to allow testing the code.